transitions/registration.js

```diff
@@ -181,59 +181,39 @@ module.exports = {
         });
     },
     triggers: {
-        add_patient_id: function(db, doc, cb) {
-            var args = Array.prototype.slice.call(arguments),
-                self = module.exports;
-            cb = args.pop();
-            if (typeof cb !== 'function') {
-                return;
-            }
+        add_patient_id: function(options, cb) {
             // if we already have a patient id then return
-            if (doc.patient_id) {
+            if (options.doc.patient_id) {
                 return;
             }
-            self.setId({db: db, doc: doc}, cb);
+            module.exports.setId(options, cb);
```

Comment:
Again, I totally agree. I'd rather leave it for now until we get around to cleaning up sentinel code more thoroughly. There are many many things I'd change if I start pulling that thread now.